core/execution/pegged_orders.go

```diff
@@ -17,65 +17,70 @@ import (
 	"sort"
 
 	"code.vegaprotocol.io/vega/core/events"
-	"code.vegaprotocol.io/vega/core/matching"
 	"code.vegaprotocol.io/vega/core/types"
 	"code.vegaprotocol.io/vega/libs/num"
+	"code.vegaprotocol.io/vega/logging"
 )
 
 type PeggedOrders struct {
+	log         *logging.Logger
 	timeService TimeService
-	orders      []*types.Order
+	// mapping orderid -> partyid
+	orders map[string]string
 
-	ordersChanged bool
+	// parked list
+	parked   []*types.Order
+	isParked map[string]struct{}
 }
 
-func NewPeggedOrders(ts TimeService) *PeggedOrders {
+func NewPeggedOrders(log *logging.Logger, ts TimeService) *PeggedOrders {
 	return &PeggedOrders{
+		log:         log,
 		timeService: ts,
-		orders:      []*types.Order{},
+		orders:      map[string]string{},
+		parked:      []*types.Order{},
+		isParked:    map[string]struct{}{},
 	}
 }
 
-func NewPeggedOrdersFromSnapshot(orders []*types.Order, tm TimeService) *PeggedOrders {
-	return &PeggedOrders{
-		timeService: tm,
-		orders:      orders,
-	}
-}
-
-// ReconcileWithOrderBook ensures that any pegged orders that are on the book point to the same
-// underlying value.
-func (p *PeggedOrders) ReconcileWithOrderBook(orderbook *matching.CachedOrderBook) error {
-	newPeggedOrders := make([]*types.Order, 0, len(p.orders))
-	for _, o := range p.orders {
-		if o.Status == types.OrderStatusParked {
-			newPeggedOrders = append(newPeggedOrders, o)
-			continue
-		}
-
-		order, err := orderbook.GetOrderByID(o.ID)
-		if err != nil {
-			return err // if its not parked it should be on the book
-		}
-		newPeggedOrders = append(newPeggedOrders, order)
+func NewPeggedOrdersFromSnapshot(
+	log *logging.Logger,
+	ts TimeService,
+	state *types.PeggedOrdersState,
+) *PeggedOrders {
+	p := NewPeggedOrders(log, ts)
+	p.orders = state.Orders
+	p.parked = state.Parked
+	for _, v := range p.parked {
+		p.isParked[v.ID] = struct{}{}
 	}
-	p.orders = newPeggedOrders
-	return nil
+	return p
 }
 
 func (p *PeggedOrders) Changed() bool {
-	return p.ordersChanged
+	return true
```

Comment:
I believe always returning true here would result in every market being re-serialised and re-hashed for each snapshot/checkpoint. Most likely (with these being set far enough apart) this will be the case anyway, but you never know... a "slow" market in auction may not have any state changed during a long auction

Reply:
orders are not saved in snapshots and now we're always serialising the whole thing for snapshot anyways. I should start removing these changed flags and functions.

Reply:
correct.